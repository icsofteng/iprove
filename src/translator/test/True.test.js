const {translate} = require('../../translator/z3')
const {translate: translate_latex} = require('../../translator/latex')

const test_constants = [{ value: 'P', varType: 'Bool' }]
const test_rules = [{
  type: 'binary',
  symbol: 'implies',
  lhs: {
    type: 'literal',
    value: 'P'
  },
  rhs: {
    type: 'true'
  }
}]


test('True test', () => {
  expect(translate(test_rules, test_constants, [], [], [])).toMatchSnapshot()
})

test('True test latex', () => {
  expect(translate_latex(test_rules, test_constants, [], [], [])).toMatchSnapshot()
})