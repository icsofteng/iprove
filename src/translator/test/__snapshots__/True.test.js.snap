// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`True test 1`] = `
"(declare-const P Bool)
(assert (not (=> P true)))
(check-sat)"
`;

exports[`True test latex 1`] = `
Array [
  "P \\\\Longrightarrow \\\\top",
]
`;
